src/test/java/echoserver/EchoServerServiceRunnableTest.java

```diff
@@ -0,0 +1,82 @@
+package echoserver;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class EchoServerServiceRunnableTest {
+
+    private EchoServerServiceRunnable serverThread;
+    private FakeClientConnection client;
+
+    private ArrayList<StringAsBytes> inputs = new ArrayList<StringAsBytes>();
+    private ByteArrayOutputStream screenOutput;
+    private Logger echoLogger;
+
+    private String clientInput = "client input";
+    private FakeSocket clientSocket;
+
+    class FakeClientConnection extends ClientConnection{
```

Comment:
Is there a reason you chose to extend `ClientConnection` instead of implementing `ConnectionDataStream`?

Reply:
I thought I took that out